src/main/java/nextstep/subway/line/domain/Line.java

```diff
@@ -68,9 +70,32 @@ public void addSection(Section section, Line line) {
 	}
 
 	public boolean isImpossibleRemoveSection() {
-		if(this.sections.getSections().size() == MIN_SECTION_COUNT){
-			return true;
-		}
-		return false;
+		return this.sections.getSections().size() == MIN_SECTION_COUNT;
+	}
+
+	public void removeTerminal(Section removedSection) {
+		boolean isUpTerminal = isUpTerminal(removedSection);
+		Section updatedSection = getUpdateSection(removedSection, isUpTerminal);
+		updatedSection.updateToTerminal(isUpTerminal);
+		this.getLineSections().remove(removedSection);
+	}
+
+	private boolean isUpTerminal(Section removedSection) {
+		return Objects.isNull(removedSection.getUpStation());
+	}
+
+	private Section getUpdateSection(Section removedSection, boolean isUpTerminal) {
+		Station linkedStation =	removedSection.getUpdateSection(isUpTerminal);
+		Optional<Section> stationOptional = this.getLineSections().stream().filter(section -> section.getMainStation().getId() == linkedStation.getId()).findAny();
+		return stationOptional.orElseThrow(() -> new IllegalArgumentException("해당 구간이 없습니다 id=" + linkedStation.getId()));
+	}
+
+	public void removeBetweenSection(Section removedSection) {
+		Section upSection = this.getLineSections().stream().filter(section -> section.getMainStation().getId() == removedSection.getUpStation().getId()).findAny().orElseThrow(() -> new IllegalArgumentException("해당 구간이 없습니다 id=" + removedSection.getUpStation().getId()));
+		Section downSection = this.getLineSections().stream().filter(section -> section.getMainStation().getId() == removedSection.getDownStation().getId()).findAny().orElseThrow(() -> new IllegalArgumentException("해당 구간이 없습니다 id=" + removedSection.getDownStation().getId()));
```

Comment:
Stream의 체이닝 길어지면 가독성이 떨어지기 때문에 아래와 같이 줄바꿈처리를 해주시면 가독성을 향상시킬 수 있을 것 같아요~!

```suggestion
		Section upSection = this.getLineSections().stream()
				.filter(section -> section.getMainStation().getId() == removedSection.getUpStation().getId())
				.findAny()
				.orElseThrow(() -> new IllegalArgumentException("해당 구간이 없습니다 id=" + removedSection.getUpStation().getId()));
		Section downSection = this.getLineSections().stream()
				.filter(section -> section.getMainStation().getId() == removedSection.getDownStation().getId())
				.findAny()
				.orElseThrow(() -> new IllegalArgumentException("해당 구간이 없습니다 id=" + removedSection.getDownStation().getId()));
```


그리고 예외에 특정값을 포함하고 싶다면 아래와 같이 커스텀 예외를 만들어 사용하면 어떨까요?

``` java
public class NotFoundSectionException extends RuntimeException {

	public NotFoundSectionException(Station station) {
	    super("해당 구간이 없습니다 id=" + station.getId());
	}
}
```